internal/internal_test.go

```diff
@@ -1596,11 +1584,27 @@ func testAuthzServerWithTruncatedBody(customLogger plugins.Plugin, dryRun, skipR
 			not input.truncated_body
 		}
 		`
-	return testAuthzServerWithModule(module, "envoy/authz/allow", customLogger, dryRun, skipRequestBodyParse)
+	return testAuthzServerWithModule(module, "envoy/authz/allow", customLogger, customConfig)
+}
+
+func TestAllowWithPrometheusMetrics(t *testing.T) {
+	var req ext_authz.CheckRequest
+	if err := util.Unmarshal([]byte(exampleAllowedRequest), &req); err != nil {
+		panic(err)
+	}
+
+	server := testAuthzServer(&testPlugin{}, &Config{EnablePrometheusMetrics: true})
+	output, err := server.Check(context.Background(), &req)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if output.Status.Code != int32(code.Code_OK) {
+		t.Fatal("Expected request to be allowed but got:", output)
+	}
```

Comment:
I don't know if the test scaffolding is capable enough -- but could we check that the desired prometheus metric is actually exported, and has a non-zero value after this incoming request?

Reply:
@srenatus done

Reply:
Done